Add BQ Loader 1.5.0

oguzhanunlu opened the pull request.

stanch commented on docs/pipeline-components-and-applications/loaders-storage-targets/bigquery-loader/snowplow-bigquery-loader-configuration-reference/index.md:
```diff
@@ -41,18 +41,22 @@ This is a complete list of the options that can be configured in the Snowplow Bi
 The defaults should be good for the overwhelming majority of deployments and hopefully you should never need to change these.
 
-| loader.loadMode.*                   | BigQuery supports two loading APIs:- Streaming inserts API- Load jobs APIThis setting configures which one will be used.StreamLoader only supports the Streaming inserts API. Loader supports both but using the Load jobs API has experimental status.                                                                                                       |
+| `loader.loadMode.*`                   | BigQuery supports two loading APIs:- Streaming inserts API- Load jobs APIThis setting configures which one will be used.StreamLoader only supports the Streaming inserts API. Loader supports both but using the Load jobs API has experimental status.                                                                                                       |
```
```suggestion
| parameter | description |
| `loader.loadMode.*`                   | BigQuery supports two loading APIs:- Streaming inserts API- Load jobs APIThis setting configures which one will be used.StreamLoader only supports the Streaming inserts API. Loader supports both but using the Load jobs API has experimental status.                                                                                                       |
```

stanch commented on docs/pipeline-components-and-applications/loaders-storage-targets/bigquery-loader/snowplow-bigquery-loader-configuration-reference/index.md:
```diff
@@ -41,18 +41,22 @@ This is a complete list of the options that can be configured in the Snowplow Bi
+| `loader.consumerSettings.awaitTerminatePeriod` | Defaults to `10 seconds`. If the underlying PubSub subcriber fails to terminate cleanly, how long do we wait until it's forcibly timed out |
+| `loader.sinkSettings.good.*`          | Settings for the good sink value in the StreamLoader code. For more details see here. For recommended number of records in each request, see here. For the HTTP request size limit, see here.   |
+| `loader.sinkSettings.bad.*`           | Settings for the bad sink value in the StreamLoader code. For more details see here. |
+| `loader.sinkSettings.types.*`         | Settings for the type sink value in the StreamLoader code. For more details see here. |
```
Here where? :)

stanch commented on docs/pipeline-components-and-applications/loaders-storage-targets/bigquery-loader/snowplow-bigquery-loader-configuration-reference/index.md:
```diff
@@ -41,18 +41,22 @@ This is a complete list of the options that can be configured in the Snowplow Bi
+| `loader.sinkSettings.bad.*`           | Settings for the bad sink value in the StreamLoader code. For more details see here. |
+| `loader.sinkSettings.types.*`         | Settings for the type sink value in the StreamLoader code. For more details see here. |
+| `loader.sinkSettings.failedInserts.*` | Settings for the failed insert sink value in the StreamLoader code. For more details see here. |
+| `loader.retrySettings.*`              | Retry settings for the BigQuery client. For more details see here. |
```
Where?

stanch commented:
There are some links in the configuration page that got lost during the migration, more specifically in a34c18b0807e439136699784263044a3265e945d.

I’ve searched for them using this: `git show a34c18b0807e439136699784263044a3265e945d~1:docs/migrated/pipeline-components-and-applications/loaders-storage-targets/bigquery-loader/snowplow-bigquery-loader-configuration-reference/index.md | grep href | pandoc -f html -t markdown | grep -B 1 ']('`

Including here in the form of this awkward table, could you pick the links from here and copy into the file? Sorry about this.

```
  ------------------------------ --------------------------------------------------------------------------------------------------------------------------------------------
  monitoring.statsd.\*           Optional. If set up, metrics will be emitted from StreamLoader and Repeater using the [StatsD](https://github.com/statsd/statsd) protocol.
--
  monitoring.statsd.prefix       Optional, eg `snowplow.monitoring`.
  monitoring.dropwizard.\*       Optional. If set up, metrics will be emitted from Loader using the [Dropwizard](https://www.dropwizard.io/en/latest/) protocol.
--
  monitoring.stdout.\*           Optional. If set up, metrics will be logged to `stdout` at `INFO` level.
  monitoring.sentry              Optional. If set up, errors will be sent to a [Sentry](https://sentry.io/) endpoint.
--
  loader.loadMode.\*                     BigQuery supports two loading APIs:\
                                         - [Streaming inserts API](https://cloud.google.com/bigquery/streaming-data-into-bigquery)\
                                         - [Load jobs API](https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs)\
--
  loader.consumerSettings.\*             Settings for the `PubsubGoogleConsumer` object in the StreamLoader code. For more details see
                                         [here](https://github.com/permutive/fs2-google-pubsub/blob/v0.15.0/fs2-google-pubsub-grpc/src/main/scala/com/permutive/pubsub/consumer/grpc/PubsubGoogleConsumerConfig.scala).
--
  loader.sinkSettings.good.\*            Settings for the good sink value in the StreamLoader code. For more details see
                                         [here](https://github.com/snowplow-incubator/snowplow-bigquery-loader/blob/master/modules/streamloader/src/main/scala/com/snowplowanalytics/snowplow/storage/bigquery/streamloader/Resources.scala#L143-L175).
                                         For recommended number of records in each request, see [here](https://cloud.google.com/bigquery/quotas#streaming_inserts). For the HTTP request size limit, see
                                         [here](https://cloud.google.com/bigquery/quotas#streaminginserts).
--
  loader.sinkSettings.bad.\*             Settings for the bad sink value in the StreamLoader code. For more details see
                                         [here](https://github.com/snowplow-incubator/snowplow-bigquery-loader/blob/master/modules/streamloader/src/main/scala/com/snowplowanalytics/snowplow/storage/bigquery/streamloader/Resources.scala#L107-L124).
--
  loader.sinkSettings.types.\*           Settings for the type sink value in the StreamLoader code. For more details see
                                         [here](https://github.com/snowplow-incubator/snowplow-bigquery-loader/blob/master/modules/streamloader/src/main/scala/com/snowplowanalytics/snowplow/storage/bigquery/streamloader/Resources.scala#L126-L141).
--
  loader.sinkSettings.failedInserts.\*   Settings for the failed insert sink value in the StreamLoader code. For more details see
                                         [here](https://github.com/snowplow-incubator/snowplow-bigquery-loader/blob/master/modules/streamloader/src/main/scala/com/snowplowanalytics/snowplow/storage/bigquery/streamloader/Resources.scala#L88-L103).

  loader.retrySettings.\*                Retry settings for the BigQuery client. For more details see [here](https://cloud.google.com/java/docs/reference/gax/latest/com.google.api.gax.retrying.RetrySettings).
```

oguzhanunlu commented on docs/pipeline-components-and-applications/loaders-storage-targets/bigquery-loader/index.md:
```diff
@@ -226,86 +231,86 @@ Mutator has three subcommands: `listen`, `create` and `add-column`.
+$ docker pull snowplow/snowplow-bigquery-repeater:${versions.bqLoader}
+`}</CodeBlock>
+
+We also provide an alternative lightweight set of images based on [Google's "distroless" base image](https://github.com/GoogleContainerTools/distroless), which may provide some security advantages for carrying fewer dependencies. These images are distinguished with the `${versions.bqLoader}-distroless` tag:
```
ah, good catch, thanks, will fix

oguzhanunlu commented:
hey @stanch , thanks for grabbing the links, I addressed your feedback in a separate commit after rebasing on latest main, could you please take a look? Thanks in advance
